cli/src/context.ts

```diff
@@ -126,3 +131,53 @@ function parseHatchFlag(args: string[]): string | boolean {
   // Otherwise, just set it to true
   return true;
 }
+
+/**
+ * Parses the target directory from command line arguments.
+ *
+ * @param args - An array of command line arguments.
+ * @returns
+ *   - `undefined` if the last argument isn't a directory path.
+ *   - The relative target path as a `string`.
+ *
+ * @description Checks if the last argument is a valid directory:
+ *
+ * 1. If no arguments, or last argument is not a target directory, returns `undefined`.
+ * 2. If last argument is invalid target (non-relative path), logs error and exits.
+ * 3. If valid target exists but has contents, requests confirmation before continuing.
+ *
+ * @example
+ * parseTargetDirectory([]) // Returns `undefined`
+ * parseTargetDirectory(["--any-flag"]) // Returns `undefined`
+ * parseTargetDirectory(["/invalid-target"]) // Logs error and exits
+ * parseTargetDirectory(["./valid-target", "--any-flag"]) // Returns `undefined`
+ * parseTargetDirectory(["./valid-target"]) // Returns "./valid-target"
+ * parseTargetDirectory(["--any-flag", "./valid-target"]) // Returns "./valid-target"
+ */
+async function parseTargetDirectory(args: string[]) {
+  const target = args.at(-1);
+
+  // `.js` check ignores `process.execPath`
+  // https://nodejs.org/docs/latest/api/process.html#processargv
+  if (!target || target.startsWith("-") || target.endsWith(".js")) {
+    return undefined;
+  }
+
+  if (target[0] !== ".") {
```

Comment:
for cross-platform support might be best to check with `path.isAbsolute`

you could also normalize the path before working with it, then still use this check (`path.normalize`)